source/windows/bcrypt_aes.c

```diff
@@ -205,6 +205,8 @@ static int s_initialize_cipher_materials(
             /* windows handles this, just go ahead and tell the API it's got a length. */
             cipher->working_mac_buffer.len = AWS_AES_256_CIPHER_BLOCK_SIZE;
         }
+    } else {
+        aws_byte_buf_reset(&cipher->working_mac_buffer, false);
```

Comment:
I'm confused what code path leads to `cipher->working_mac_buffer` being uninitialized?

The cipher is always allocated like: `aes_bcrypt_cipher *cipher = aws_mem_calloc(...)` which zeroes out everything.

And if the cipher is reset, it ends up calling [aws_byte_buf_secure_zero(&cipher_impl->working_mac_buffer)](https://github.com/awslabs/aws-c-cal/blob/82f64a749954dd3371b41768c4bd790406e09146/source/windows/bcrypt_aes.c#L280-L281)

So how did this buffer end up uninitialized?

Reply:
I'm just pushing back on the location of the fix.

Typically, we try to initialize/zero-out everything to start, so we don't have to worry about uninitialized memory later

Having a fix for uninitialized memory so far down in a function is a fragile place to put it.

Reply:
yeah for sure, I will change its location.
These testcases fail on ARM64 for windows
 ```
         75 - aes_cbc_NIST_CBCGFSbox256_case_1 (Failed)
         76 - aes_cbc_NIST_CBCVarKey256_case_254 (Failed)
         77 - aes_cbc_NIST_CBCVarTxt256_case_110 (Failed)
         78 - aes_cbc_NIST_CBCMMT256_case_4 (Failed)
         79 - aes_cbc_NIST_CBCMMT256_case_9 (Failed)
         80 - aes_cbc_test_with_generated_key_iv (Failed)
         82 - aes_ctr_RFC3686_Case_7 (Failed)
         83 - aes_ctr_RFC3686_Case_8 (Failed)
         84 - aes_ctr_RFC3686_Case_9 (Failed)
         85 - aes_ctr_test_with_generated_key_iv (Failed)
         96 - aes_keywrap_RFC3394_256BitKey256CekTestVector (Failed)
         97 - aes_keywrap_Rfc3394_256BitKey_TestIntegrityCheckFailed (Failed)
         98 - aes_keywrap_RFC3394_256BitKeyTestBadPayload (Failed)
         99 - aes_keywrap_RFC3394_256BitKey128BitCekTestVector (Failed)
        100 - aes_keywrap_RFC3394_256BitKey128BitCekIntegrityCheckFailedTestVector (Failed)
        101 - aes_keywrap_RFC3394_256BitKey128BitCekPayloadCheckFailedTestVector (Failed)
```
After the fix they all passed.
This is a stack trace for the failure.
```
To run a test, enter the test number: 74
Fatal error condition occurred in C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\crt\aws-crt-cpp\crt\aws-c-common\source\byte_buf.c:88: aws_byte_buf_is_valid(buf)
Exiting Application
at 0x7FF6E3197B60: aws_backtrace(C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\crt\aws-crt-cpp\crt\aws-c-common\source\windows\system_info.c:162)
at 0x7FF6E3197FB8: aws_backtrace_print(C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\crt\aws-crt-cpp\crt\aws-c-common\source\windows\system_info.c:250)
at 0x7FF6E3183B08: aws_fatal_assert(C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\crt\aws-crt-cpp\crt\aws-c-common\source\assert.c:17)
at 0x7FF6E318C478: aws_byte_buf_secure_zero(C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\crt\aws-crt-cpp\crt\aws-c-common\source\byte_buf.c:89)
at 0x7FF6E317C7C8: s_initialize_cipher_materials(C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\crt\aws-crt-cpp\crt\aws-c-cal\source\windows\bcrypt_aes.c:216)
at 0x7FF6E317CE80: s_reset_cbc_cipher(C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\crt\aws-crt-cpp\crt\aws-c-cal\source\windows\bcrypt_aes.c:295)
at 0x7FF6E31701B8: aws_symmetric_cipher_reset(C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\crt\aws-crt-cpp\crt\aws-c-cal\source\symmetric_cipher.c:206)
at 0x7FF6E3119A68: s_check_single_block_cbc(C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\crt\aws-crt-cpp\crt\aws-c-cal\tests\aes256_test.c:31)
at 0x7FF6E311A6A8: s_NIST_CBCGFSbox256_case_1_fn(C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\crt\aws-crt-cpp\crt\aws-c-cal\tests\aes256_test.c:64)
at 0x7FF6E31195C0: s_aws_run_test_case(C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\out\install\arm64-Debug\include\aws\testing\aws_test_harness.h:461)
at 0x7FF6E311A8E0: aes_cbc_NIST_CBCGFSbox256_case_1(C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\crt\aws-crt-cpp\crt\aws-c-cal\tests\aes256_test.c:66)
at 0x7FF6E3118638: main(C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\crt\aws-crt-cpp\crt\aws-c-cal\out\build\arm64-Debug\tests\test_runner.c:836)
at 0x7FF6E31CF228: invoke_main(D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78)
at 0x7FF6E31CED38: __scrt_common_main_seh(D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
at 0x7FF6E31CECD0: __scrt_common_main(D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:330)
at 0x7FF6E31CF420: mainCRTStartup(D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_main.cpp:16)
at 0x7FFC80C922E0: BaseThreadInitThunk
at 0x7FFC84935740: RtlUserThreadStart
PS C:\Users\alfre\workspace\aws-iot-device-sdk-cpp-v2\crt\aws-crt-cpp\crt\aws-c-cal\out\build\arm64-Debug\tests> .\aws-c-cal-tests.exe
```

Reply:
looking at the function you mentioned `aws_mem_calloc`, i see that it doesn't support ARM processors on windows
I will fix it